### DISCUSSION TITLE
FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD for cert

### DISCUSSION THREAD

**Discussion by burtsevyg** [fastlane/fastlane#17993] [Q&A]
It's unclear in documentation can I use apple specific password for cert action?
And if it is impossible how to sigh build ipa file which I want to upload to testflight?
Should I use spaceauth?

My Fastfile:

```
default_platform(:ios)

platform :ios do
  desc "Build ipa file"
  lane :build_ipa do
    cert
    sigh(
      force: true,
      app_identifier: "my_app_identifier"
    )
    update_code_signing_settings(
      use_automatic_signing: true
    )
    gym(
      workspace: "Runner.xcworkspace",
      scheme: "Runner"
    )
  end
end

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :deploy do
    upload_to_testflight(skip_waiting_for_build_processing: true)
  end
end

```

If I remove valid cookie
`/Users/user/.fastlane/spaceship/*/cookie`
 and add FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD I get: 
`Please enter the 6 digit code:`

on `cert` action not on `upload_to_testflight`

**Comment by dinsen** [comment]
Hi @burtsevyg 
I think the app specific password is only for use with ITMSTransporter (upload to TestFlight).

**Reply by burtsevyg** [reply]
Hi @Dinsen.
Thank you for your reply.
Is there any way for ios to automate recreating cookie on CI server every month or there is one way to do it by hand via spaceauth:

fastlane spaceauth -u developer@email.com
Please enter the 6 digit code:

This 6 digit I get on my phone because set SPACESHIP_2FA_SMS_DEFAULT_PHONE_NUMBER variable.

**Reply by dinsen** [reply]
Currently there is no way to automate this.

There is this PR https://github.com/fastlane/fastlane/pull/17487
But since the creator haven't signed the CLA, it won't be reviewed 😞 

Have you considered using the API key? 
https://docs.fastlane.tools/app-store-connect-api/
